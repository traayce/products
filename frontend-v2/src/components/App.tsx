import * as React from "react";
import "./../assets/scss/App.scss";

export interface AppProps {
}

export default class App extends React.PureComponent<AppProps> {
    render() {
        return (
            <div className="app">
                <div>init</div>
            </div>
        );
    }
}
