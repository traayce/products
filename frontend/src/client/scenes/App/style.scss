.appShell {
  font-size: 16px;
  font-family: sans-serif;
  height: 100%;
}

