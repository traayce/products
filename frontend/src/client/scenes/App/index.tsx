import PrivateRoute from 'client/components/PrivateRoute';
import * as React from 'react';
import { Helmet } from 'react-helmet';
import createLazyContainer from 'react-lazy-import';
import { Route, RouteComponentProps, Switch } from 'react-router-dom';
import Footer from './components/Footer';
import Navbar from './components/Navbar';
import './normalize.scss';
import * as style from './style.scss';
const Loading = () => <div>loading</div>;
const Error = () => <div>error</div>;

const Products = createLazyContainer<RouteComponentProps<{}>>(
  () => import(
    'client/scenes/Products'
  ), Loading, Error
);
const NotFound = createLazyContainer<RouteComponentProps<{}>>(
  () => import(
    /* webpackChunkName: 'NotFound' */
    'client/scenes/NotFound'
  ), Loading, Error
);

interface IState {
  hasError: boolean;
}

class App extends React.Component<{}, IState> {
  public state: IState = {
    hasError: false
  };

  public componentDidCatch () {
    this.setState({ hasError: true });
  }

  public render () {
    if (this.state.hasError) {
      return (
        <div>
          Whoops! This app broke :(
        </div>
      );
    }

    return (
      <div style={{height: '100%'}}>
        <div className={style.appShell}>
        <Helmet
          titleTemplate='%s - Products'
          defaultTitle='Products'
        />
        <Navbar />
        <Switch>
          <Route exact path='/' component={Products} />
          <Route path='/products' component={Products} />
          <Route component={NotFound} />
        </Switch>
        </div>
        <Footer />
      </div>
    );
  }
}

export default App;
