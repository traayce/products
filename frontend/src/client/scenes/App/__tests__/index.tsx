import * as React from 'react';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';

/**
 * Node doesn't have System.import, so we have to mock it
 */
(global as any).System = {
  import (path: string) {
    return new Promise((resolve, reject) => {
      resolve({ default: () => <div>{path}</div> });
    });
  }
};

import App from '../index';


describe('App scene', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
