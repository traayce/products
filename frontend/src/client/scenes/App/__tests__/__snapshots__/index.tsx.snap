// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App scene should render correctly 1`] = `
<div
  className="appShell"
>
  <div
    className="navbarContainer"
  >
    <ul
      className="navbar"
    >
      <li
        className="item"
      >
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="item"
      >
        <a
          className="link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li
        className="item"
      >
        <a
          className="link"
          href="/protected"
          onClick={[Function]}
        >
          Protected
        </a>
      </li>
      <li
        className="item"
      >
        <a
          className="link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </div>
  <div>
    loading
  </div>
  <footer
    className="footer"
  >
    <p>
      ðŸŽ‰  Presentational App
    </p>
  </footer>
</div>
`;
